Exporter:Tolerance exceeded in error handler

現行connector 在寫出檔案時會出錯，報錯內容如下
```
2022-12-30T17:06:14.456723975Z org.apache.kafka.connect.errors.ConnectException: Tolerance exceeded in error handler
2022-12-30T17:06:14.456728580Z 	at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:223)
2022-12-30T17:06:14.456732054Z 	at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execute(RetryWithToleranceOperator.java:149)
2022-12-30T17:06:14.456735051Z 	at org.apache.kafka.connect.runtime.WorkerSinkTask.convertAndTransformRecord(WorkerSinkTask.java:513)
2022-12-30T17:06:14.456738288Z 	at org.apache.kafka.connect.runtime.WorkerSinkTask.convertMessages(WorkerSinkTask.java:493)
2022-12-30T17:06:14.456741173Z 	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:332)
2022-12-30T17:06:14.456744044Z 	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
2022-12-30T17:06:14.456746877Z 	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
2022-12-30T17:06:14.456749851Z 	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:189)
2022-12-30T17:06:14.456752669Z 	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:244)
2022-12-30T17:06:14.456755474Z 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2022-12-30T17:06:14.456758343Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-12-30T17:06:14.456762014Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2022-12-30T17:06:14.456765184Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2022-12-30T17:06:14.456768220Z 	at java.base/java.lang.Thread.run(Thread.java:829)
2022-12-30T17:06:14.456771235Z Caused by: org.apache.kafka.connect.errors.DataException: Converting byte[] to Kafka Connect data failed due to serialization error: 
2022-12-30T17:06:14.456774288Z 	at org.apache.kafka.connect.json.JsonConverter.toConnectData(JsonConverter.java:324)
2022-12-30T17:06:14.456777165Z 	at org.apache.kafka.connect.storage.Converter.toConnectData(Converter.java:88)
2022-12-30T17:06:14.456780064Z 	at org.apache.kafka.connect.runtime.WorkerSinkTask.lambda$convertAndTransformRecord$3(WorkerSinkTask.java:513)
2022-12-30T17:06:14.456794656Z 	at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndRetry(RetryWithToleranceOperator.java:173)
2022-12-30T17:06:14.456799013Z 	at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:207)
2022-12-30T17:06:14.456802176Z 	... 13 more
2022-12-30T17:06:14.456805074Z Caused by: org.apache.kafka.common.errors.SerializationException: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'test': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
2022-12-30T17:06:14.456808267Z  at [Source: (byte[])"test"; line: 1, column: 5]
2022-12-30T17:06:14.456811158Z 	at org.apache.kafka.connect.json.JsonDeserializer.deserialize(JsonDeserializer.java:66)
2022-12-30T17:06:14.456813966Z 	at org.apache.kafka.connect.json.JsonConverter.toConnectData(JsonConverter.java:322)
2022-12-30T17:06:14.456816809Z 	... 17 more
2022-12-30T17:06:14.456819564Z Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'test': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
2022-12-30T17:06:14.456822840Z  at [Source: (byte[])"test"; line: 1, column: 5]
2022-12-30T17:06:14.456825819Z 	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2391)
2022-12-30T17:06:14.456828777Z 	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:745)
2022-12-30T17:06:14.456831794Z 	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3635)
2022-12-30T17:06:14.456835216Z 	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3608)
2022-12-30T17:06:14.456838061Z 	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._matchToken2(UTF8StreamJsonParser.java:2949)
2022-12-30T17:06:14.456840894Z 	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._matchTrue(UTF8StreamJsonParser.java:2883)
2022-12-30T17:06:14.456843754Z 	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:875)
2022-12-30T17:06:14.456846840Z 	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:794)
2022-12-30T17:06:14.456849802Z 	at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4703)
2022-12-30T17:06:14.456852722Z 	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3090)
2022-12-30T17:06:14.456855633Z 	at org.apache.kafka.connect.json.JsonDeserializer.deserialize(JsonDeserializer.java:64)
2022-12-30T17:06:14.456858473Z 	... 18 more
2022-12-30T17:06:14.457431925Z 
```

--- a/docs/connector/exporter.md
+++ b/docs/connector/exporter.md
@@ -23,7 +23,7 @@
 | path                      | (選填)   填入目標檔案系統要儲存的資料夾目錄之目標位置                                                               | 無     |
 | size                      | (選填)   寫入檔案目標超過此設定之大小上限時會創見新檔案，並且寫入目標改為新創建之檔案。  <br/>檔案大小之單位可以設定為 `Bit`, `Kb`, `KiB`, `Mb`等 | 100MB |     |
 
-注意：如果`worker`預設之`converter`並非為`byte array convert`
+注意：如果`worker`預設之`converter`並非為`byte array converter`
 時，需要在將以下參數使設定為`org.apache.kafka.connect.converters.ByteArrayConverter`使本工具順利執行。
 
 - key.converter

--- a/docs/connector/exporter.md
+++ b/docs/connector/exporter.md
@@ -5,10 +5,9 @@
 輸出之檔案會依照當初在叢集中的 `partition` 來進行資料夾的分隔，每個檔案之檔名為其包含之第一筆 `record` 的 `offset`。
 
 目前支援之檔案系統有以下列出項目
+
 - local
 - ftp
-
-
 
 #### Exporter Configurations
 
@@ -24,6 +23,12 @@
 | path                      | (選填)   填入目標檔案系統要儲存的資料夾目錄之目標位置                                                               | 無     |
 | size                      | (選填)   寫入檔案目標超過此設定之大小上限時會創見新檔案，並且寫入目標改為新創建之檔案。  <br/>檔案大小之單位可以設定為 `Bit`, `Kb`, `KiB`, `Mb`等 | 100MB |     |
 
+注意：如果`worker`預設之`converter`並非為`byte array convert`
+時，需要在將以下參數使設定為`org.apache.kafka.connect.converters.ByteArrayConverter`使本工具順利執行。
+
+- key.converter
+- value.converter
+- header.converter
 
 #### 使用範例
 

--- a/docs/connector/exporter.md
+++ b/docs/connector/exporter.md
@@ -23,12 +23,29 @@
 | path                      | (選填)   填入目標檔案系統要儲存的資料夾目錄之目標位置                                                               | 無     |
 | size                      | (選填)   寫入檔案目標超過此設定之大小上限時會創見新檔案，並且寫入目標改為新創建之檔案。  <br/>檔案大小之單位可以設定為 `Bit`, `Kb`, `KiB`, `Mb`等 | 100MB |     |
 
-注意：如果`worker`預設之`converter`並非為`byte array converter`
-時，需要在將以下參數使設定為`org.apache.kafka.connect.converters.ByteArrayConverter`使本工具順利執行。
 
+##### 注意
+如果`worker`預設之`converter`並非為`byte array converter`
+時，需要在將以下參數使設定為`org.apache.kafka.connect.converters.ByteArrayConverter`使本工具順利執行。
 - key.converter
 - value.converter
 - header.converter
+
+因為當`topic`內的資料為非 `worker converter`包裝的話，沒有設定`converter`會出現類似以下的錯誤資訊
+```
+org.apache.kafka.connect.errors.ConnectException: Tolerance exceeded in error handler
+at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:223)
+...
+Caused by: org.apache.kafka.connect.errors.DataException: Converting byte[] to Kafka Connect data failed due to serialization error:
+at org.apache.kafka.connect.json.JsonConverter.toConnectData(JsonConverter.java:324)
+...
+Caused by: org.apache.kafka.common.errors.SerializationException: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'test': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
+...
+Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'test': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
+...
+```
+
+
 
 #### 使用範例
 

--- a/docs/gui/connector/README.md
+++ b/docs/gui/connector/README.md
@@ -32,6 +32,8 @@
 提醒：Kafka 預設會使用 `json convert`來處理 records，因此如果使用本專案的`backup`工具，
 需要將 `converter`設定為 `org.apache.kafka.connect.converters.ByteArrayConverter`
 
+詳情可以至[Exporter](../../connector/exporter.md#注意)查看
+
 ![connector_create_1.png](connector_create_1.png)
 
 ### active workers

--- a/docs/gui/connector/README.md
+++ b/docs/gui/connector/README.md
@@ -11,11 +11,13 @@
 - [plugin](#plugin)
 
 ### basic
+
 檢視當前`worker`所執行中的`connector`其名稱、狀態與參數內容，如需要可以在此頁面刪除`connector`
 
 ![connector_basic_1.png](connector_basic_1.png)
 
 ### task
+
 當下`worker`所執行之`task`創建資訊與其當下統計之狀態
 
 當`task`有出現`error`時，會在`error` field 展示其 exception 內容
@@ -24,16 +26,22 @@
 *因缺少`jar`檔，因此出現錯誤並顯示在 `error` field中*
 
 ### create
+
 可以透過 gui 在此 `worker`創建所需要之`connector`，創建完成時會在下方顯示創造出來的`connector`
+
+提醒：Kafka 預設會使用 `json convert`來處理 records，因此如果使用本專案的`backup`工具，
+需要將 `converter`設定為 `org.apache.kafka.connect.converters.ByteArrayConverter`
 
 ![connector_create_1.png](connector_create_1.png)
 
 ### active workers
+
 顯示當下所執行之 `worker`，包含資訊有執行`connector`,`task`數量、版本與`cluster id`等資訊
 
 ![connector_active_workers_1.png](connector_active_workers_1.png)
 
 ### plugin
+
 可以檢視此`worker`以安裝之`plugin`資訊與其文件，兩者都可以透過`query`進行搜尋與過濾
 
 ![connector_plugin_1.png](connector_plugin_1.png)

